spring-amqp/src/main/java/org/springframework/amqp/support/converter/AbstractJackson2MessageConverter.java

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright 2002-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.amqp.support.converter;
+
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.amqp.core.Message;
+import org.springframework.amqp.core.MessageProperties;
+import org.springframework.beans.factory.BeanClassLoaderAware;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.util.Assert;
+import org.springframework.util.ClassUtils;
+import org.springframework.util.MimeType;
+
+import java.io.IOException;
+import java.lang.reflect.Type;
+
+/**
+ * Abstract Jackson2 message converter
+ *
+ * @author Mohammad Hewedy
+ */
+public abstract class AbstractJackson2MessageConverter extends AbstractMessageConverter
+		implements BeanClassLoaderAware, SmartMessageConverter {
+
+	private static Log log = LogFactory.getLog(AbstractJackson2MessageConverter.class);
+
+	public static final String DEFAULT_CHARSET = "UTF-8";
+
+	private volatile String defaultCharset = DEFAULT_CHARSET;
+
+	private ClassMapper classMapper = null;
+
+	protected boolean typeMapperSet;
+
+	private final MimeType contentType;
+
+	protected final ObjectMapper objectMapper;
+
+	private ClassLoader classLoader = ClassUtils.getDefaultClassLoader();
+
+	private Jackson2JavaTypeMapper javaTypeMapper = new DefaultJackson2JavaTypeMapper();
+
+	/**
+	 * Construct with the provided {@link ObjectMapper} instance.
+	 * @param objectMapper the {@link ObjectMapper} to use.
+	 * @param contentType content type of the message
+	 * @param trustedPackages the trusted Java packages for deserialization
+	 * @since 1.6.11
+	 * @see DefaultJackson2JavaTypeMapper#setTrustedPackages(String...)
+	 */
+	public AbstractJackson2MessageConverter(ObjectMapper objectMapper, MimeType contentType, String... trustedPackages) {
```

Comment:
This ctor must be `protected`. You can't require from the target implementors to expose such a ctor.